### ISSUE TITLE
3 of 15 BTRFS system Partitions failled to backup. Logs gone to space ?..

### ISSUE THREAD

**Issue by TERIIHUB** [rescuezilla/rescuezilla#309]
Hi RZ team,

While doing a full SSD partitions backup, our final report states that 3 SYS partitions with Linux on, had a backup failure.
*checking at the Var/Log : 
*it seems that after each partition is backed-up the following log  overwrite the previous one (?) So we can only find a single short lenght file with details about the last partition process...
QUESTION : 
Is there anything we can improve to properly get a complete backup of our multiboot system and check what went on for each fail ?
Thanks.
SSD detail below ; 
---------------------------------------------------------------------
Backup Summary
Backup succeeded with some errors:
Successful backup of partition /dev/sdb1
Successful backup of partition /dev/sdb2
Successful backup of partition /dev/sdb3
<b>Failed to backup partition</b> /dev/sdb4       <<< GARUDA
Successful backup of partition /dev/sdb5
Successful backup of swap partition /dev/sdb11
Successful backup of partition /dev/sdb12
Successful backup of partition /dev/sdb13
Successful backup of partition /dev/sdb14
<b>Failed to backup partition</b> /dev/sdb18    <<< KaOs
Successful backup of partition /dev/sdb19
<b>Failed to backup partition</b> /dev/sdb20    <<<<<<
Successful backup of partition /dev/sdb21
Successful backup of partition /dev/sdb22
Successful backup of partition /dev/sdb23
Operation took 10.2 minutes.
---------------------------------------------------------------------
Partitions os SSD backed up : Sys and Data
sdb   465,8G         
sdb1    800M vfat    BOOTA
sdb11   7,8G swap    
sdb12    51G btrfs   
sdb13  34,9G btrfs   KUBUNPLAS
sdb14  24,4G btrfs   BACKBTRFS
sdb18    44G btrfs   KAOS
sdb19    44G ntfs    NTFSONE
sdb2     44G btrfs   VOYAGER
sdb20  27,3G btrfs   MANJARO
sdb21  19,5G ext4    EXTONE
sdb22  35,1G hfsplus HFPLUS500
sdb23  19,5G nilfs2  NILFONE
sdb3     44G btrfs   FERENOS
sdb4     44G btrfs   GARUDAXFCE
sdb5   25,7G btrfs   VOIDOS
--------------------------------------------------------------
Data partition are just 'fs' for testing/comparing purpose...
-----------------------------------------------
Last 'and only' Var/log found..  :O ?!

+1: Partclone v0.3.13 http://partclone.org
Starting to clone device (/dev/sdb23) to image (-)
Reading Super Block
memory needed: 2737156 bytes
bitmap 640000 bytes, blocks 2*1048576 bytes, checksum 4 bytes
Calculating bitmap... Please wait... 
done!
File system:  NILFS
Device size:   21,0 GB = 5120000 Blocks
Space in use:  25,2 MB = 6144 Blocks
Free Space:    20,9 GB = 5113856 Blocks
Block size:   4096 Byte
Total block 5120000
Syncing... OK!
Partclone successfully cloned the device (/dev/sdb23) to the image (-)
---------------------------------------------
end of report

**Comment by shasheene** [comment]
BTRFS support in partclone isn't very robust, and Rescuezilla v2.3.1 uses the version of `partclone` that's in the Debian repositories rather than compiling from source like Clonezilla does.

This will be improved in the next release. Sorry I haven't yet fixed this.

**Comment by shasheene** [comment]
The full and complete log file is available within the file `clonezilla-img` in your backup folder.

**Comment by TERIIHUB** [comment]
Hi Shasheene,
Thanks for the reply /
Right, I had read a bit about some partclone problems backuping [didnt know rescue used it beside] :

stil not sure I could locate that 'log' file [name?] within the img uncompressed folder...it only had something about a smart txt.

now, I could grab the messagebox content it poped each time it fails though, so i ll just past it here below if that can be of use :
anyway... 
      Is ther any plan to use a 2nd option tool such as fsarchiver ...beside partclone as a workarround ?
I'm running test with some tools and, though fsarchiver builds a one file, that can only be browsed when uncompressed, I'd be looking more into a 'one click and forget' backup solution if that one can do the job. 

I'm currently testing : luckybackup / rsync / Fsa / cp . 
Also *btfrs pickup is a recent choice to move away from ext4...
 if you'd think another fs would not pose such problems with RZ, I d be interested to know which one/s ? :   

**any idea with theses partitions in particular and not others  ?

Returned msg : A
*********************************************
Failed to backup partition</b> /dev/sdb20
The command used internally was:
partclone.btrfs --logfile /var/log/partclone.log --clone --source /dev/sdb20 --output - | cat - | split --suffix-length=2 --bytes=4GB - /mnt/backup/SSDcompletTENTATIVE2/2022-02-22-1259-img-rescuezilla/sdb20.btrfs-ptcl-img.uncomp. 
The output of the command was: Partclone v0.3.13 http://partclone.org
Starting to clone device (/dev/sdb20) to image (-)
Reading Super Block
couldn't open because of unsupported option features (10).
btrfsclone.c: Couldn't open file system
Partclone fail, please check /var/log/partclone.log !
******************************************
B
******************************************
The command used internally was:
partclone.btrfs --logfile /var/log/partclone.log --clone --source /dev/sdb18 --output - | cat - | split --suffix-length=2 --bytes=4GB - /mnt/backup/SSDcompletTENTATIVE2/2022-02-22-1259-img-rescuezilla/sdb18.btrfs-ptcl-img.uncomp. 
The output of the command was: Partclone v0.3.13 http://partclone.org
Starting to clone device (/dev/sdb18) to image (-)
Reading Super Block
couldn't open because of unsupported option features (10).
btrfsclone.c: Couldn't open file system
Partclone fail, please check /var/log/partclone.log !<b>Failed to backup partition</b> /dev/sdb18
The command used internally was:
partclone.btrfs --logfile /var/log/partclone.log --clone --source /dev/sdb18 --output - | cat - | split --suffix-length=2 --bytes=4GB - /mnt/backup/SSDcompletTENTATIVE2/2022-02-22-1259-img-rescuezilla/sdb18.btrfs-ptcl-img.uncomp. 
The output of the command was: Partclone v0.3.13 http://partclone.org
Starting to clone device (/dev/sdb18) to image (-)
Reading Super Block
couldn't open because of unsupported option features (10).
btrfsclone.c: Couldn't open file system
Partclone fail, please check /var/log/partclone.log !
**************************************************
C
************************************************
The command used internally was:
partclone.btrfs --logfile /var/log/partclone.log --clone --source /dev/sdb4 --output - | cat - | split --suffix-length=2 --bytes=4GB - /mnt/backup/SSDcompletTENTATIVE2/2022-02-22-1259-img-rescuezilla/sdb4.btrfs-ptcl-img.uncomp. 
The output of the command was: Partclone v0.3.13 http://partclone.org
Starting to clone device (/dev/sdb4) to image (-)
Reading Super Block
couldn't open because of unsupported option features (10).
btrfsclone.c: Couldn't open file system
Partclone fail, please check /var/log/partclone.log 
***************************************

**Comment by shasheene** [comment]
Linux's widely used EXT4 filesystem is very well supported by `partclone`.

Hopefully updated versions of `partclone` improve BTRFS support though.

**Comment by TERIIHUB** [comment]
Ok, yeah, good to know anyway..hoping hoping :)
wondering what clonezilla uses now...tks,
have a good dev week !

**Comment by shasheene** [comment]
Clonezilla uses `partclone` but a more recent version compiled from source. Have you had success with Clonezilla on this machine?

I'll try and get Rescuezilla v2.4 released this month.

**Comment by TERIIHUB** [comment]
not yet, I only have used CZ years ago on a pro level and NTFS fs only config, so not since then...
I will run test with it soon...
As of now i'm just learning about UrBackup as a client/server option [although it comes without Gui 4 linux]
Cool, I ll keep an eye on RZ v2.4 anyway,,, :]

**Comment by TERIIHUB** [comment]
okay, yeah,,well, this Urbackup is quite time consuming to setup. I could install the server side but the client part is not very user friendly with plenty of params to understands and etc...in otherwords thats the last complex setup i'd be looking for some emergency disaster recovey case on a client machine, maybe its easy onces everything is set up...but not my cup of coffe at this time.

So, back to the classical tools : CZ. clonezilla-live-2.8.1-12-i686
As for CZ, standard Full Disk with various Btrfs partitions to a single image img (comression/default) did not seemed to return any bckuping error, Op concluded in 50min to a single img.
img was auto checked with no defaults

As a reverse way..., Img back to restoring some Btrfs partitions to disk :
I simulated a 'fake' restoring process [without writtings indeed] which also seemed to be working without any restoring errors for that much...not real physical test/writes were done,
So all in all, this version of CZ seems to be the only 'practical' way to backup your sys partitions.
rescuezilla was not able to browse teh CZ archive or work with it.

So, not many options left here...

Another options to get a system back- if it prooves to be even more time efficient - would be to use Nix to plan for a full sys, reinstallation. it maybe as efficient if not even much much quicker than trying to restore some image in some case...:) 
That's next on line with clonezilla as a Plan B option.

**Comment by shasheene** [comment]
That's for the insight! Yeah not being able to browse a standard Clonezilla archive using "Image Explorer (beta)" is expected right now, because you'll need to create it as uncompressed for it to work with Rescuezilla.

Interesting on the other points. It certainly sounds like the new partclone version will solve the BTRFS issues in Rescuezilla, given it works in Clonezilla. I'll get it done.

**Comment by TERIIHUB** [comment]
Tested today a full 4 Btrfs sys partitions + boot from img to ssd, everything seems fine. 
So as of now, Clonezilla is working fine with btrfs. not sure I do have a choice though for the Compression, CZ offers 2 options and the first one is not clear what compression it uses or if z=0, none.

**Comment by shasheene** [comment]
With Clonezilla, I believe you have to go into Expert mode to set 'no compression'. Beginner mode only shows a few options and I believe 'no compression' may not be one of them.

**Comment by TERIIHUB** [comment]
Correct, I've just used it now, working fine again,,, thanks for that precision... :)

**Comment by TERIIHUB** [comment]
HI,
I've just posted to the CZ forum, CZ has started to fail me sincs yesterday,
...dont really know why or if becauze I added a new ext4 partition among all btrfs...
will wait for reply from some bigboss :)
![photo_2022-03-24_16-58-41](https://user-images.githubusercontent.com/3276014/160048826-5a690f2c-1016-4b5b-97a1-f9efe7cfee3b.jpg)
![photo_2022-03-24_16-58-47](https://user-images.githubusercontent.com/3276014/160048872-47ad935a-9515-43da-98d0-4cb4212c9c87.jpg)

**Comment by shasheene** [comment]
Sounds like you may be using an old version of Clonezilla which doesn't have the most recent version of partclone?

**Comment by TERIIHUB** [comment]
correct, i've switched to using a previous version [completing the backup but with an error warning]...
 after the latest version was not completing the backup process anymore... and freezing / photos above.

**Comment by TERIIHUB** [comment]
hey, I've been switching from Ext4 to Btrfs and finally to xfs..
this configuration seems to backup decently using rescuezilla !
:)

**Comment by shasheene** [comment]
Update: the latest version of `partclone` is now being built from source. It's available in Rescuezilla "Weekly" Rolling Release (2022-07-24) which will form the basis of Rescuezilla v2.4.

This fixes the compressed BTRFS issue.

**Comment by TERIIHUB** [comment]
Cool, Thanks, I 'll post that to some contact who are using btrfs now (I'm happy with xfs finallly), thanks for the great work !
